routes/project/location/kubernetes_cluster.rb

```diff
@@ -27,6 +27,17 @@ class Clover
           kc.incr_destroy
           204
         end
+
+        r.on "kubeconfig" do
+          r.get do
+            authorize("KubernetesCluster:create", kc.id)
```

Comment:
This permission seems incorrect.  For one this is a read-only request, not a request to change anything.  Second, in other places `KubernetesCluster:create` is checked against the project's id, not the id of the cluster.  `KubernetesCluster:view` is a read-only permission, but you may not want to grant everyone with access to view the cluster to download the config.  If you don't want to add a specific permission for this `KubernetesCluster:edit` or `KubernetesCluster:delete` is probably better, as that is designed to check against the cluster id.

Reply:
Thanks for catching this. I think edit makes sense now, maybe we can include another permission specific for kubeconfig too..